packages/flutter/lib/src/widgets/modal_barrier.dart

```diff
@@ -14,6 +14,97 @@ import 'gesture_detector.dart';
 import 'navigator.dart';
 import 'transitions.dart';
 
+/// A widget that modifies the size of the [SemanticsNode.rect] (focus) of its
+/// child widget. It clips the focus in possibily four directions based on the
+/// specified [EdgeInsets].
+///
+/// If a [ValueNotifier] is provided, the size of the focus is adjusted based on
+/// value changes inside the [ValueNotifier].
+///
+/// If no [ValueNotifier] is provided, the [SemanticsClipper] applies a default
+/// value of [EdgeInsets.zero], which preserves the default size of the focus.
+///
+/// See also:
+///
+///  * [ModalBarrier], which utilizes this widget to adjust the barrier focus
+/// size based on the size of the content layer rendered on top of it.
+class _SemanticsClipper extends SingleChildRenderObjectWidget{
+  /// creates a [SemanticsClipper] that updates the size of the
+  /// [SemanticsNode.rect] of its child based on the value inside the provided
+  /// [ValueNotifier], or a default value of [EdgeInsets.zero].
+  const _SemanticsClipper({
+    super.child,
+    this.clipDetailsNotifier,
+  });
+
+  /// The [ValueNotifier] whose value determines how the child's
+  /// [SemanticsNode.rect] should be clipped in four directions.
+  final ValueNotifier<EdgeInsets>? clipDetailsNotifier;
+
+  @override
+  _RenderSemanticsClipper createRenderObject(BuildContext context) {
+    return _RenderSemanticsClipper(clipDetailsNotifier: clipDetailsNotifier,);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, _RenderSemanticsClipper renderObject) {
+    renderObject.clipDetailsNotifier = clipDetailsNotifier;
+  }
+}
+/// Updates the [SemanticsNode.rect] of its child based on the value inside
+/// provided [ValueNotifier].
+/// If no [ValueNotifier] is provided a value of [EdgeInsets.zero] is used
+/// to clip the [SemanticsNode.rect], which essentially uses its default size.
+class _RenderSemanticsClipper extends RenderProxyBox {
+/// Creats a [RenderProxyBox] that Updates the [SemanticsNode.rect] of its child
+/// based on the value inside provided [ValueNotifier].
+  _RenderSemanticsClipper({
+    ValueNotifier<EdgeInsets>? clipDetailsNotifier,
+    RenderBox? child,
+  }) : _clipDetailsNotifier = clipDetailsNotifier,
+       super(child);
+
+  ValueNotifier<EdgeInsets>? _clipDetailsNotifier;
+  /// The getter and setter retrieves / updates the [ValueNotifier] associated
+  /// with this clipper.
+  ValueNotifier<EdgeInsets>? get clipDetailsNotifier => _clipDetailsNotifier;
+  set clipDetailsNotifier (ValueNotifier<EdgeInsets>? newNotifier) {
+    if (_clipDetailsNotifier == newNotifier) {
+      return;
+    }
+    _clipDetailsNotifier = newNotifier;
```

Comment:
you will need to check if `attached` or not, if it is true, you need to removeListener on old `_clipDetailsNotifier` and addListener on the newNotifier

Reply:
Thanks! Done